mopidy/file/library.py

```diff
@@ -82,7 +82,14 @@ def browse(self, uri):
             elif os.path.isfile(child_path):
                 result.append(models.Ref.track(name=name, uri=uri))
 
-        result.sort(key=operator.attrgetter('name'))
+        sort_funcs = [operator.attrgetter('name'),
+                      lambda item: 0 if item.type == models.Ref.DIRECTORY
+                      else 1]
+        logger.debug('sort_funcs %s', sort_funcs)
+
+        for key_func in sort_funcs:
```

Comment:
> Doing it like this I you will end up first sorting it just by name, and then resorting it just by type == models.Ref.DIRECTORY.

Yes - this leads to the desired effect (first directories sorted by name, then files sorted by name), due to the python's stable sorting algorithm (not changing the order of "equal" items in successive runs).

> On a side note int(True) -> 1 and int(False) -> 0.

Yes, but I think the natural order of boolean values is not as obvious for the reader as the natural order of integer numbers. A matter of taste :)

Merging the successive sorting iterations into a single sorting run is reasonable, since the sorting priorities are not supposed to be configurable anymore.